### ISSUE TITLE
Cannot connect telepresence for some reason...

### ISSUE THREAD

**Issue by aaron-trout** [telepresenceio/telepresence#1320]
### What were you trying to do?
Run `telepresence` with no args (except context/namespace) to access a service in the cluster.

### What did you expect to happen?

I expected it to work 😆

### What happened instead?

Python exception happened and didn't connect

### Automatically included information

Command line: `['/usr/local/bin/telepresence', '--context', 'gke-production', '--namespace', 'default']`
Version: `0.104`
Python version: `3.7.7 (default, Mar 10 2020, 15:43:33) 
[Clang 11.0.0 (clang-1100.0.33.17)]`
kubectl version: `Client Version: v1.18.1 // Server Version: v1.15.11-eks-af3caf`
oc version: `(error: [Errno 2] No such file or directory: 'oc': 'oc')`
OS: `Darwin Aarons-MacBook-Pro-4.local 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64`

```
Traceback (most recent call last):
  File "/usr/local/bin/telepresence/telepresence/cli.py", line 135, in crash_reporting
    yield
  File "/usr/local/bin/telepresence/telepresence/main.py", line 65, in main
    remote_info = start_proxy(runner)
  File "/usr/local/bin/telepresence/telepresence/proxy/__init__.py", line 138, in start_proxy
    run_id=run_id,
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 144, in get_remote_info
    runner, deployment_name, deployment_type, run_id=run_id
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 91, in get_deployment_json
    return json.loads(output)["items"][0]
IndexError: list index out of range

```

Logs:

```
 --ignore-not-found svc,deploy --selector=telepresence=c422e01af1e64d76b1a99ddc19176c2c
  20.4   9 | No resources found
  20.4 TEL | [9] ran in 0.17 secs.
  20.4 TEL | [10] Running: kubectl --context gke-production --namespace default run --restart=Always --limits=cpu=1000m,memory=256Mi --requests=cpu=25m,memory=64Mi telepresence-1587548770-759708-70638 --image=datawire/telepresence-k8s:0.104 --labels=telepresence=c422e01af1e64d76b1a99ddc19176c2c
  20.5  10 | pod/telepresence-1587548770-759708-70638 created
  20.5 TEL | [10] ran in 0.17 secs.
  20.5 TEL | END SPAN deployment.py:108(create_new_deployment)    0.3s
  20.5 TEL | BEGIN SPAN remote.py:142(get_remote_info)
  20.5 TEL | BEGIN SPAN remote.py:75(get_deployment_json)
  20.5 TEL | [11] Capturing: kubectl --context gke-production --namespace default get deployment -o json --selector=telepresence=c422e01af1e64d76b1a99ddc19176c2c
  20.7 TEL | [11] captured in 0.12 secs.
  20.7 TEL | END SPAN remote.py:75(get_deployment_json)    0.1s

```

This is happening every time I try to connect across GKE and EKS clusters.

**Comment by aaron-trout** [comment]
Also worth noting that the `kubectl version` in the "automatically collected information" for the bug report is wrong, since it does not respect the `--context` flag, so the server version is for whatever my default context is, not the one I specified in the `telepresence` command.

**Comment by marun** [comment]
@aaron-trout Duplicate of #1297, which was fixed by #1312. Should be resolved in the next release, and until then #1297 documents a number of workarounds.
